scripts/benchmark_invoice_list_performance.rb

```
@@ -0,0 +1,156 @@
+    puts "   Average: #{(times.sum / times.length).round(2)}ms"
+    puts "   Min:     #{times.min}ms"
+    puts "   Max:     #{times.max}ms"
+    puts "   Median:  #{times.sort[times.length / 2]}ms"
+  end
+
+  def explain_analyze_query
+    puts "\n🔍 Query execution plan:"
+    
+    sql = build_invoice_list_sql
+    result = ActiveRecord::Base.connection.execute("EXPLAIN (ANALYZE, BUFFERS) #{sql}")
+    
```

RuboCop (1.75.5)

[convention] Trailing whitespace detected.

(Layout/TrailingWhitespace)